website/blog/2023-01-04-ferretdb-beta-version-authentication.md

```diff
@@ -0,0 +1,117 @@
+description: The FerretDB beta version (v.0.8.0) includes exciting new features, including authentication for PostgreSQL, `$min` operator support, and much more.
+author: Alexander Fashakin
+author_title: FerretDB Team
+---
+
+![FerretDB 0.8.0 release](../static/img/blog/FerretDB-is-now-Beta.-1-980x551.png)
+
+<!--truncate-->
+
+FerretDB - the open source alternative to MongoDB - is thrilled to announce release of our  Beta version(0.8.0), which includes various new features, bug fixes, improved documentation, and, most importantly, the implementation of authentication for PostgreSQL.
+This new and exciting release is no ordinary milestone; it's a culmination of FerretDB's journey as we work on bringing you the ultimate open-source alternative to MongoDB by converting MongoDB protocol queries to SQL, with PostgreSQL as the database engine.
+
+While we do not aim to cover all the features of MongoDB, our goal with the Beta - and our upcoming GA release later this quarter - is to provide a solid foundation on which to build targeted features that'll enable FerretDB to support more and more real-world use cases.
+Be sure to [check our roadmap for further details](https://github.com/orgs/FerretDB/projects/2).
+
+:::caution
+Please note that this particular change breaks backward compatibility.
+To dump and restore the data, you’ll need connection strings to connect to FerretDB and PostgreSQL.
+There are numerous ways to dump and restore your data.
+For example, you can follow the following steps:
+
+1. Backup FerretDB databases with `mongodump`.
+Set your FerretDB connection string in `—uri` and run:
+   `mongodump --uri="mongodb://localhost:27017"`
+   This command will create a directory with dumps for each FerretDB database on the given server.
+Later, we will use this dump to restore the database.
+Instead of `mongodump`, you can also use `mongoexport` for this.
+
+2. Backup PostgreSQL `ferretdb` database.
+If the migration goes well, we will not use this backup.
+But we recommend to do this step in case you need to rollback.
+Specify your host and port.
+`pg_dump -h localhost -p 5432 -U username ferretdb > ferretdb.sql`
+
+3. Stop FerretDB (This depends on your operating system and the way you run FerretDB)
+
+4. Connect to PostgreSQL and drop `ferredb` database as it's not needed anymore:
+   * `psql -h localhost -p 5432 -U username postgres`
+   * `DROP DATABASE ferretdb`
+
+5. Upgrade FerretDB and run FerretDB 0.8 (Please refer to our documentation where we describe how to update and start FerretDB)
+
+6. Restore database using `mongorestore --uri="mongodb://localhost:27017"`
+
+7. In case something doesn't work and you need to rollback to FerretDB 0.7.1:
+   * Stop ferretdb
+   * Delete ferretdb database (repeat step 4)
+   * Restore `ferretdb` PostgreSQL db from the dump we created on the step 2:  `psql -h localhost -p 5432 -U username ferretdb -f ferretdb.sql`
+   * Start ferretdb 0.7.1
+:::
```

Comment:
In principle, in this case it should be possible (and better) to rollback with `mongorestore` too. Postgres' backup is a backup-backup, in case if something went really wrong.

Reply:
Is this something we want to add? "You can rollback with `mongorestore`. However, in case something doesn't work..." and then continue from there

Reply:
Yes, that sounds good to me!